docs/cmake-presets.md

```diff
@@ -0,0 +1,272 @@
+
+This article contains information about `CMakePresets.json` integration in the CMake Tools extension for Visual Studio Code.
+- For more information on the format of `CMakePresets.json`, see the official [CMake Tools documentation](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html). 
+- For more information on the Microsoft vendor maps and macro expansion, see [`CMakePresets.json` and `CMakeUserPresets.json` Microsoft vendor maps](https://docs.microsoft.com/cpp/build/cmake-presets-json-reference).
+- For more information on how to use `CMakePresets.json` in Visual Studio, see [Configure and build with CMake presets in Visual Studio](https://docs.microsoft.com/cpp/build/cmake-presets-vs)
+
+`CMakePresets.json` is a recommended alternative to kits and variants files. See [Enable `CMakePresets.json` in the CMake Tools extension](#enable-cmakepresetsjson-in-the-cmake-tools-extension) to enable or disable `CMakePresets.json` integration in the CMake Tools extension.
+
+## Supported CMake and `CMakePresets.json` versions
+
+The CMake Tools extension supports `CMakePresets.json` and `CMakeUserPresets.json` files version 2 or higher. You can update your file version by incrementing the version field in the root object. For an example and more information, see [`CMakePresets.json`](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html#format).
+
+CMake version 3.20 or higher is required when invoking CMake with `CMakePresets.json` (version 2 or higher) from the command line. CMake Tools reads and evaluates `CMakePresets.json` and `CMakeUserPresets.json`, and does not invoke CMake directly with the `--preset` option. This means CMake version 3.20 or higher is not strictly required when building with `CMakePresets.json` inside Visual Studio Code. Use at least CMake version 3.14 or higher.
+
+## Enable `CMakePresets.json` in the CMake Tools extension
+
+A new setting, `cmake.useCMakePresets`, has been added to `settings.json`:
+
+|Setting  |Description  |Accepted values  | Default value |
+|---------|---------|---------|---------|
+|`cmake.useCMakePresets` |  Drive CMake configure, build, and test | `true`, `false`, `auto` | `auto` |
+
+`auto` : evaluates to `true` if there's a `CMakePresets.json` file in the `cmake.sourceDirectory` in the active folder. It evaluates to `false` if there isn't a `CMakePresets.json` file in the `cmake.sourceDirectory` in the active folder. Set `cmake.useCMakePresets` to `true` or `false` to explicitly enable or disable `CMakePresets.json` integration for all CMake projects.
+
+## Configure and build
+
+You can configure and build your CMake project with a series of commands. Open the command palette in Visual Studio Code with `Ctrl+Shift+P`:
+
+![Command pallette: Select Configure Preset](images/command-palette.png)
```

Comment:
I'm not sure if this is rendering correctly (at least in the preview)

Reply:
I don't believe images render in preview out here. Looks ok when I render it locally. I'll keep an eye out for this.